pex/resolver.py

```diff
@@ -314,14 +276,8 @@ def from_request(
         )
         return cls(request=build_request, atomic_dir=AtomicDirectory(dist_dir))
 
-    def __init__(
-        self,
-        request,  # type: BuildRequest
-        atomic_dir,  # type: AtomicDirectory
-    ):
-        # type: (...) -> None
-        self.request = request
-        self._atomic_dir = atomic_dir
+    request = attr.ib()  # type: BuildRequest
+    _atomic_dir = attr.ib()  # type: AtomicDirectory
```

Comment:
N.B.: attrs is smart enough to make an underscored attribute available by its non-underscored name as a kwarg. Publically constructible via kwarg, "private thereafter".